### PR TITLE
[CAS-234] New CAS3 API for reports functionality

### PR DESCRIPTION
**Pull request by sudhakar-plr**
The PR contains changes to 
1. Create new report API `/cas3/reports/{reportType}`
2. Refactor existing report service to support new and old report API

Note: Once the UI change to use new API, we will remove the existing API and its implementation.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/properties/TransitionalAccommodationReferralReportProperties.kt [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
+import java.time.LocalDate
 import java.util.UUID
 
 data class TransitionalAccommodationReferralReportProperties(
-  val serviceName: ServiceName,
   val probationRegionId: UUID?,
-  val year: Int,
-  val month: Int,
+  val startDate: LocalDate,
+  val endDate: LocalDate,
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/properties/TransitionalAccommodationReferralReportProperties.kt:9]: i like moving to actual dates, makes things far more flexible!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReportsController.kt [hunk 2]
```diff
@@ -24,12 +24,15 @@ import uk.gov.justice.digital.hmpps.approvedpremisesapi.reporting.properties.Ref
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.ReportService
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.UserAccessService
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.UserService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas3.ReportServiceForCas3
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReportsController.kt:27]: for consistency with other services could we please call this the Cas3ReportService? (we have lots of Cas1... classes)

**Review by davidatkinsuk — APPROVED** [review]
LGTM, just a few small proposals

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReportsController.kt (hunk 2 above)
Comment (sudhakar-plr) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReportsController.kt:27]: I thought about it, but prefer the serviceName first approach. I can see the value for consistency as well.

File: src/main/resources/static/_shared.yml [hunk 3]
```diff
@@ -4463,6 +4463,13 @@ components:
           type: string
       required:
         - name
+    ReportType:
```
[the remark this answers is not in the document]
Comment (sudhakar-plr) [reply src/main/resources/static/_shared.yml:4466]: I didnt see CAS2 changes before, so it make sense to specific to CAS3.

[the remark this answers is not in the document]
Comment (sudhakar-plr) [reply src/main/resources/static/_shared.yml]: done

**Pull request merged**